### PR TITLE
re-introducing column-view

### PR DESCRIPTION
**Pull request by TyJOrtiz**
code for column-view and pages

**Resolved / Related Issues**

Itemize resolved / related issues by this PR.
Added code relevant to Column View. 

**Details of Changes**

Add details of changes here.
- added Column View to LayoutModes enums
- added ColumnView seitch commands

**Validation**

How did you test these changes?
- [X] Built and ran the app

**Screenshots (optional)**

Add screenshots here.
image

### CODE REVIEW

**Comment by bl-ue** [comment]
Re-implementation of #2500

File: Files/ViewModels/FolderSettingsViewModel.cs [hunk 1]
```diff
@@ -172,6 +175,35 @@ public void SwitchAdaptiveLayout(bool enable)
             LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, GridViewSize));
         });
 
+        public RelayCommand<bool> ToggleLayoutModeColumnView => new RelayCommand<bool>((manuallySet) =>
+        {
+            if (App.AppSettings.AreLayoutPreferencesPerFolder && App.AppSettings.AdaptiveLayoutEnabled)
+            {
+                if (LastLayoutModeSelected == FolderLayout.GridViewLarge && LayoutPreference.IsAdaptiveLayoutOverridden)
```
Comment (d2dyno1) [inline Files/ViewModels/FolderSettingsViewModel.cs:182]:
```cs
LastLayoutModeSelected == FolderLayout.ColumnView
```

File: Files/ViewModels/FolderSettingsViewModel.cs [hunk 2]
```diff
@@ -172,6 +175,35 @@ public void SwitchAdaptiveLayout(bool enable)
...
+                {
+                    SwitchAdaptiveLayout(true);
+                    return;
+                }
+                else if (LastLayoutModeSelected == FolderLayout.GridViewSmall || LastLayoutModeSelected == FolderLayout.GridViewMedium || LastLayoutModeSelected == FolderLayout.GridViewLarge)
```
Comment (d2dyno1) [inline Files/ViewModels/FolderSettingsViewModel.cs:187]: Remove this `else if`

File: Files/ViewModels/FolderSettingsViewModel.cs [hunk 3]
```diff
@@ -172,6 +175,35 @@ public void SwitchAdaptiveLayout(bool enable)
...
+                    // Override preferred gridview size
+                }
+                else if (manuallySet)
+                {
+                    // Override preferred layout mode
+                    SwitchAdaptiveLayout(false);
+                }
+            }
+
+            LayoutMode = FolderLayoutModes.ColumnView; // Grid View
```
Comment (d2dyno1) [inline Files/ViewModels/FolderSettingsViewModel.cs:198]:
```cs
// Column View
```

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

File: Files/ViewModels/FolderSettingsViewModel.cs [hunk 4]
```diff
@@ -172,6 +175,35 @@ public void SwitchAdaptiveLayout(bool enable)
...
+
+            GridViewSize = Constants.Browser.GridViewBrowser.GridViewSizeLarge; // Size
```
Comment (d2dyno1) [inline Files/ViewModels/FolderSettingsViewModel.cs:200]: Remove please

File: Files/ViewModels/FolderSettingsViewModel.cs [hunk 5]
```diff
@@ -172,6 +175,35 @@ public void SwitchAdaptiveLayout(bool enable)
...
+
+            LastLayoutModeSelected = FolderLayout.ColumnView;
+
+            LayoutModeChangeRequested?.Invoke(this, new LayoutModeEventArgs(LayoutMode, 0));
```
Comment (d2dyno1) [inline Files/ViewModels/FolderSettingsViewModel.cs:204]: new LayoutModeEventArgs(LayoutMode, GridViewSize)

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 6]
```diff
@@ -147,6 +147,15 @@
                             <FontIcon  Glyph="&#xE179;" />
                         </MenuFlyoutItem.Icon>
                     </MenuFlyoutItem>
+                    <MenuFlyoutItem
+                    x:Uid="BaseLayoutContextFlyoutColumn"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:151]: Can you add this string to the en-US resource file?

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/Views/LayoutModes/GenericFileBrowser.xaml (hunk 6 above)
Comment (TyJOrtiz) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:151]: done

**Review by d2dyno1 — DISMISSED** [review]

File: Files/Views/LayoutModes/GenericFileBrowser.xaml (hunk 6 above)
Comment (d2dyno1) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:151]: @yaichenbaum

**Review by TyJOrtiz — COMMENTED** [review]
👍

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 7]
```diff
@@ -866,8 +875,7 @@
                 </Grid>
                 <Popup
                     x:Name="EditPopup"
-                    Grid.Row="1"
-                    x:Load="False">
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:870]: Can you restore `x:Load`?

Comment (TyJOrtiz) [reply Files/Views/LayoutModes/GridViewBrowser.xaml:870]: 👍

**Review by yaichenbaum — APPROVED** [review]

**Comment by yaichenbaum** [comment]
@TyJOrtiz I am merging this with the intention that we will continue working on it and make sure everything is working before releasing it to users.

**Comment by bl-ue** [comment]
🚀 🎊

**Comment by TyJOrtiz** [comment]
Ok cool.

There's like only a couple things that I need help with to make it ready to ship.

**Comment by yaichenbaum** [comment]
@TyJOrtiz That's great! Can you open issues so we can keep track of what still needs to be done?

**Comment by TyJOrtiz** [comment]
How do I reopen this PR

**Comment by yaichenbaum** [comment]
@TyJOrtiz I think you need to open a new one since this was already merged.

**Comment by TyJOrtiz** [comment]
Done

Sent from my iPhone

On Mar 31, 2021, at 9:33 AM, Yair Aichenbaum ***@***.***> wrote:

﻿

@TyJOrtiz<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FTyJOrtiz&data=04%7C01%7C%7C8eb32bf3959d4a54ad1b08d8f462b273%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637528052025072212%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=WGvQJ9JUOoU%2FvdHu%2FKgm2ALn92MWIGroPKmmqgh%2BfDM%3D&reserved=0> I think you need to open a new one since this was already merged.

**Pull request merged**